datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaConnector.java

```diff
@@ -463,6 +487,38 @@ public String reduce(String query, Map<String, String> responses) {
     return null;
   }
 
+  private String reduceTopicPartitionStatsResponses(Map<String, String> responses) {
+    Map<String, Map<String, Set<Integer>>> result = new HashMap<>();
+
+    responses.forEach((instance, json) -> {
+      List<KafkaTopicPartitionStatsResponse> responseList;
+      try {
+        responseList = KafkaTopicPartitionStatsResponse.fromJson(json);
+      } catch (Exception e) {
+        _logger.error("Invalid response {} from instance {}", json, instance);
+        return;
+      }
+      if (responseList.size() >= 0) {
+        responseList.forEach(response -> {
+          if (response.getTopicPartitions() == null || response.getConsumerGroupId().isEmpty()) {
+            _logger.warn("Empty topic partition stats map from instance {}. Ignoring the result", instance);
+            return;
```

Comment:
should we throw exception here so that caller of the endpoint will know about it?

Reply:
I think ignoring the result and proceeding further is good here.  At least we will return a warning and proceed further with best effort. This is a diag tool.